src/github/client/app-token-holder.ts

```diff
@@ -5,14 +5,35 @@ import { AuthToken, ONE_MINUTE, TEN_MINUTES } from "./auth-token";
 import * as PrivateKey from "probot/lib/private-key";
 import LRUCache from "lru-cache";
 import { InstallationId } from "./installation-id";
+import { Subscription } from "~/src/models/subscription";
+import { GitHubServerApp } from "~/src/models/github-server-app";
+import { booleanFlag, BooleanFlags } from "~/src/config/feature-flags";
 
-export type KeyLocator = (installationId: InstallationId) => string;
+export type KeyLocator = (installationId: InstallationId) => Promise<string>;
```

Comment:
I don't think we need this type at all, we just need the function to be typed correctly

Reply:
@mboudreau, it is fetching private key from table which is async operation. How do you want to type this one?

Reply:
I'm just talking about the type itself.  I don't think we should be using it as it doesn't need to be passed in.  We just need to call the function directly

Reply:
ie. have it in imports, we don't need to inject it into the constructor as we should be using the same one everywhere.

Reply:
What's the downside of having this function as a type? I find that types make the code much smoother to read (i.e. I know that it's a key locator when I see the type).

Reply:
@thombergs all functions are typed, I'm saying that we don't need an abstract of that function type because we don't need to inject it into the constructor in the first place.  We should be able to call it directly since our key locator should be universal.